src/components/Dashboard.js

```diff
@@ -37,6 +38,8 @@ export const Dashboard = () => {
   const onChangeValue = (value) => {
     if (regex.test(value.target.value))
       setUserValue(value.target.value)
+    if (!value.target.value)
+      setUserValue(null)
```

Comment:
would this invoke `onChangeValue` method again?

Reply:
another thing - we should check if there is value at first. And it is - check regexp. This way script will check only necessary conditions. 
Maybe use else if? 
```
if (!value.target.value) { 
  setUserValue(null) 
} else if (regex.test(value.target.value)) { 
  setUserValue(value.target.value
} 
```

Or just add `return` to prevent method from additional checking.

Reply:
> would this invoke `onChangeValue` method again?

It wouldn't

Reply:
Fixed